syntactic-whitelist/src/main/scala/Features.scala

```diff
@@ -0,0 +1,353 @@
+    case _ : Term.While => true
+    case _ : Pat.Var => true
+  })
+
+  /**
+   * Allows the use of contextual constructs, including:
+   * implicits, using, givens
+   */
+  case object AllowContextualConstructs extends AtomicFeature({
+    case Mod.Implicit() => true
+    case Mod.Using() => true
+    case _ : Decl.Given => true
+    case _ : Defn.Given => true
+    case _ : Defn.GivenAlias => true
+    case _ : Importee.Given => true
+    case _ : Importee.GivenAll => true
+    case _ : Pat.Given => true
+    case _ : Type.ImplicitFunction => true
+    case _ : Term.ApplyUsing => true
+  })
+
+  /**
+   * Allows the definition of extension groups and extension methods
+   */
+  case object AllowExtensions extends AtomicFeature({
+    case _ : Defn.ExtensionGroup => true
+  })
+
+  /**
+   * Allows the use of macro and metaprogramming-related constructs
+   */
+  case object AllowMetaprogramming extends AtomicFeature({
+    case _ : Defn.Macro => true
+    case _ : Term.QuotedMacroExpr => true
+    case _ : Term.QuotedMacroType => true
+    case _ : Term.SplicedMacroExpr => true
+    case _ : Term.SplicedMacroPat => true
+    case _ : Type.Macro => true
+    case _ : Pat.Macro => true
+    case _ : internal.trees.Quasi => true
+  })
+
+  /**
+   * Allows the use of packages
+   */
+  case object AllowPackages extends AtomicFeature({
+    case _ : Pkg => true
+    case _ : Pkg.Object => true
+  })
+
+  /**
+   * Allows the use of imports
+   */
+  case object AllowImports extends AtomicFeature({
+    case _ : Import => true
+    case _ : Importee => true
+    case _ : Importer => true
+    case _ : Name.Indeterminate => true
+  })
+
+  /**
+   * Allows export clauses
+   */
+  case object AllowExports extends AtomicFeature({
+    case _ : Export => true
+    case _ : Importee => true
+    case _ : Importer => true
+    case _ : Name.Indeterminate => true
+  })
+
+  /**
+   * Allows the use of XML-related constructs
+   */
+  case object AllowXml extends AtomicFeature({
+    case _ : Term.Xml => true
+    case _ : Pat.Xml => true
+    case _ : Lit.String => true
+  })
+
+  /**
+   * Allows string interpolation
+   */
+  case object AllowStringInterpolation extends AtomicFeature({
+    case _ : Term.Interpolate => true
+    case _ : Pat.Interpolate => true
+  })
+
+  /**
+   * Allows annotations
+   */
+  case object AllowAnnotations extends AtomicFeature({
+    case _ : Term.Annotate => true
+    case _ : Mod.Annot => true
+  })
+
+  /**
+   * Allows to define infix methods
+   */
+  case object AllowInfixes extends AtomicFeature({
+    case Mod.Infix() => true
+  })
+
+  /**
+   * Allows to define inline methods
+   */
+  case object AllowInlines extends AtomicFeature({
+    case Mod.Inline() => true
+  })
+
+  /*
+  Not implemented  TODO figure out what they are
+  Member.Term
+  Member.Type
+  MultiSource
+  Term.ApplyType
+  Term.FunctionTerm
+  Term.Ref
+   */
+
+  val ALL_FEATURES: List[Feature] = List(
```

Comment:
No, at least I don't see any. But I thought that it could be useful to have a list of all the features that are defined in `Features.scala`, that is why I created this constant.